File: lib/src/data/models/client/client_log.dart
import 'package:json_annotation/json_annotation.dart';

part 'client_log.g.dart';

@JsonSerializable()
class ClientLog {
  const ClientLog({required this.level, required this.message});

  /// 日志级别
  /// 0=Panic, 1=Fatal, 2=Error, 3=Warn, 4=Info, 5=Debug, 6=Trace
  final int level;
  final String message;

  factory ClientLog.fromJson(Map<String, dynamic> json) => _$ClientLogFromJson(json);

  Map<String, dynamic> toJson() => _$ClientLogToJson(this);
}

File: android/src/main/kotlin/com/clashsing/flutter_sing_box/cs/models/ClientLog.kt
package com.clashsing.flutter_sing_box.cs.models

import android.annotation.SuppressLint
import kotlinx.serialization.Serializable

/**
  * @property level 日志级别 (0=Panic, 1=Fatal, 2=Error, 3=Warn, 4=Info, 5=Debug, 6=Trace)
 *
 */
@SuppressLint("UnsafeOptInUsageError")
@Serializable
data class ClientLog(
    val level: Int,
    val message: String
)
